hantec_api_ecommerce/controllers/main.py

```diff
@@ -0,0 +1,915 @@
+
+        JSON response:
+            - message (str): Confirmation message with the ID of the scheduled activity and the invoice ID.
+
+        Returns:
+            dict: Dictionary with a confirmation message.
+        """
+        data = request.jsonrequest
+        invoice_id = data.get("invoice_id")  # ID of the existing invoice
+        activity_type_id = data.get("activity_type_id")  # ID of the activity type
+        summary = data.get("summary", "")  # Summary text of the activity (optional)
+        date_deadline = data.get("date_deadline")  # Deadline date for the activity
+        note = data.get("note", "")  # Note for the activity (optional)
+        user_id = data.get("user_id", request.env.uid)  # User assigned to the activity
+
+        # Create the scheduled activity
+        activity = request.env["mail.activity"].create(
+            {
+                "activity_type_id": activity_type_id,
+                "note": note,
+                "date_deadline": date_deadline,
+                "res_model_id": request.env["ir.model"]._get("account.move").id,
+                "res_id": invoice_id,
+                "user_id": user_id,
+                "summary": summary,
+            }
+        )
+        logger.info(
+            "Scheduled activity created with ID %s for invoice %s",
+            activity.id,
+            invoice_id,
+        )
+
+        return {
+            "message": f"Scheduled activity created with ID: {activity.id} for invoice {invoice_id}."
+        }
+
+    @route("/send_message_sale_order", methods=["POST"], type="json", auth="user")
+    def send_message_sale_order(self):
+        """
+        Sends a message to a specific sale order.
+
+        This function posts a message to the chatter of a sale order identified by its ID,
+        using the message body provided in the JSON request body.
+
+        JSON request body:
+            - sale_order_id (int): The ID of the sale order.
+            - message_body (str): The content of the message to be posted.
+
+        JSON response:
+            - message (str): A confirmation message indicating the action performed.
+
+        Returns:
+            dict: A dictionary with a confirmation message.
+
+        """
+        data = request.jsonrequest
+        sale_order_id = data.get("sale_order_id")
+        message_body = data.get("message_body")
+
+        # Find the sale order by its ID
+        sale_order = request.env["sale.order"].browse(sale_order_id)
+
+        # Post the message in the sale order chatter
+        sale_order.message_post(body=message_body)
+        logger.info("Message posted in sale order with ID %s", sale_order_id)
+
+        return {
+            "message": f"Message successfully posted in sale order with ID: {sale_order_id}."
+        }
+
+    @route("/get_inventory", methods=["GET"], type="json", auth="user")
```

Comment:
merge this controller with the one below

just add another route in for the same method